File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/01-lab/04.sumOfTwoNumbers.js
function sumOfTwoNumbers(input) {

    const startNum = Number(input[0]);
    const endNum = Number(input[1]);
    const magicNum = Number(input[2]);

    let counterCombinations = 0;
    let isCombination = false;

    for (let i = startNum; i <= endNum; i++) {
        for (let j = startNum; j <= endNum; j++) {
            counterCombinations++;
            if (i + j === magicNum) {
                console.log(`Combination N:${counterCombinations} (${i} + ${j} = ${magicNum})`);
                isCombination = true;
                break;
            }
        }

        if (isCombination === true) {
            break;
        }
    }

    if (!isCombination) {
        console.log(`${counterCombinations} combinations - neither equals ${magicNum}`);
    }
}
sumOfTwoNumbers(["88", "888", "1000"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/01.lab/04.sum_of_two_numbers.py
start_num = int(input())
end_num = int(input())
magic_num = int(input())

counter_combinations = 0
is_combination = False

for i in range(start_num, end_num + 1):
    for j in range(start_num, end_num + 1):
        counter_combinations += 1
        if i + j == magic_num:
            print(f'Combination N:{counter_combinations} ({i} + {j} = {magic_num})')
            is_combination = True
            break
            
    if is_combination == True:
        break
        
if not is_combination:
    print(f'{counter_combinations} combinations - neither equals {magic_num}')
